orchagent/vxlanorch.cpp

```diff
@@ -385,27 +397,57 @@ remove_tunnel_termination(sai_object_id_t term_table_id)
     }
 }
 
-bool VxlanTunnel::createTunnel(MAP_T encap, MAP_T decap, uint8_t encap_ttl)
+bool VxlanTunnel::createTunnel(vector<MAP_T> encap, vector<MAP_T> decap, uint8_t encap_ttl)
 {
     try
     {
         sai_ip_address_t ips, ipd, *ip=nullptr;
         swss::copy(ips, src_ip_);
 
-        ids_.tunnel_decap_id = SAI_NULL_OBJECT_ID;
-        ids_.tunnel_encap_id = SAI_NULL_OBJECT_ID;
+        std::vector<sai_object_id_t> decap_list;
+        std::vector<sai_object_id_t> encap_list;
 
-        if (decap != MAP_T::MAP_TO_INVALID)
+        for (auto d : decap)
         {
-            ids_.tunnel_decap_id = create_tunnel_map(decap);
+            if (d == MAP_T::VNI_TO_VLAN_ID)
+            {
+                ids_.tunnel_vni_to_vlan_decap_id = create_tunnel_map(d);
+                decap_list.push_back(ids_.tunnel_vni_to_vlan_decap_id);
+            }
+            else if (d == MAP_T::VNI_TO_VRID)
+            {
+                ids_.tunnel_vni_to_vrid_decap_id = create_tunnel_map(d);
+                decap_list.push_back(ids_.tunnel_vni_to_vrid_decap_id);
+            }
+            else if (d == MAP_T::VNI_TO_BRIDGE)
+            {
+                ids_.tunnel_vni_to_bridge_decap_id = create_tunnel_map(d);
+                decap_list.push_back(ids_.tunnel_vni_to_bridge_decap_id);
+            }
         }
-        if (encap != MAP_T::MAP_TO_INVALID)
+
+        for (auto e : encap)
         {
-            ids_.tunnel_encap_id = create_tunnel_map(encap);
+            if (e == MAP_T::VLAN_ID_TO_VNI)
```

Comment:
Avoid using ```if else```, according to the another comment):
https://github.com/volodymyrsamotiy/sonic-swss/pull/2/files#r624532442
> Use ```map``` to store mappers IDs in order to optimize the way how it is stored and also to avoid many ```if``` statements in a code that uses these variables.